ecolex/legislation.py

```diff
@@ -251,32 +265,46 @@ def add_legislations(legislations, count_ignored):
                 new_legislations.append(legislation)
                 new_docs.append(doc)
         except SolrError as e:
-            logger.error('Error importing legislation %s' % (leg_id,))
+            logger.error(f'Error importing legislation {leg_id}')
             failed_docs.append(doc)
             if settings.DEBUG:
                 logger.exception(e)
 
         doc.parsed_data = json.dumps(legislation)
-        if (doc.url != legislation['legLinkToFullText']):
-            doc.url = legislation['legLinkToFullText']
+        if (doc.url != legislation.get('legLinkToFullText')):
+            doc.url = legislation.get('legLinkToFullText')
             # will not re-parse if same url and doc_size
             doc.doc_size = None
+        doc.save()
 
     # Mix of exceptions and True/False return status
     # hoping that add_bulk has better performance than add
     count_new = index_and_log(solr, new_legislations, new_docs)
     count_updated = index_and_log(solr, updated_legislations, updated_docs)
+
+    for doc in failed_docs:
+        importer.reindex_failed(doc)
+
+    logger.info(f'[Legislation] Update full text started.')
     for doc in new_docs + updated_docs + failed_docs:
         doc.save()
+        importer.update_full_text(doc)
+    logger.info(f'[Legislation] Update full text finished.')
+
+    logger.info(f'Total {len(legislations) + count_ignored}. '
+                f'Added {count_new}. Updated {count_updated}. '
+                f'Failed {len(legislations) - count_new - count_updated}. '
+                f'Ignored {count_ignored}')
+
+def _set_values_from_dict2(data, field, local_dict):
```

Comment:
I'm not very happy with the naming of this function; what is the difference between this function and `_set_values_from_dict`?